docs/developers/source-code-declspec.rst

```diff
@@ -0,0 +1,31 @@
+Source code: Symbol visibility
+==============================
+
+The `*_DECLSPEC` macros provide a method to annotate symbols to indicate
+their intended visibility when compiling dynamically shared object files
+(e.g., `libmpi.so`).
+
+The macros are defined on a per project basis:
+
+    * Open MPI: `OMPI_DECLSPEC`, `OMPI_MODULE_DECLSPEC`
+    * Open PAL: `OPAL_DECLSPEC`, `OPAL_MODULE_DECLSPEC`
+    * OpenSHMEM: `OSHMEM_DECLSPEC`, `OSHMEM_MODULE_DECLSPEC`
+
+The macros expand to the appropriate compiler and platform flags for marking
+whether a symbol should be explicitly made public in the target projects
+library namespace.  This is related to the Windows [declspec](https://docs.microsoft.com/en-us/cpp/cpp/declspec?view=msvc-170).
+
+The `*_DECLSPEC` attribute is used to declare that the developer wants a
```

Comment:
Use double quotes for fixed-width font (same for other instances throughout this file):

```suggestion
The ``*_DECLSPEC`` attribute is used to declare that the developer wants a
```